core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -515,6 +515,11 @@ class ReplicaManager(val config: KafkaConfig,
             case Some(p) =>
               if (p eq ReplicaManager.OfflinePartition)
                 throw new KafkaStorageException("Partition %s is in an offline log directory on broker %d".format(topicPartition, localBrokerId))
+              p.leaderReplicaIfLocal match {
+                case Some(_) =>
+                case None =>
+                  throw new NotLeaderForPartitionException("Broker %d is not leader for partition %s".format(localBrokerId, topicPartition))
```

Comment:
I understand that using Scala template strings (`s"Broker $localBrikerId is not leader..."`) is preferred over `String.format()`

Reply:
Also, long line, but using templates string might fix that.